packages/core/src/fhircast/test-utils.ts

```diff
@@ -0,0 +1,26 @@
+import { FhircastEventContext, FhircastResourceType } from '.';
+
+const FHIRCAST_RESOURCE_TYPES = {
+  patient: 'Patient',
+  imagingstudy: 'ImagingStudy',
+  encounter: 'Encounter',
+} as const;
+
+export function createFhircastMessageContext(
+  resourceType: Lowercase<FhircastResourceType>,
+  resourceId: string
+): FhircastEventContext {
+  if (!FHIRCAST_RESOURCE_TYPES[resourceType]) {
+    throw new TypeError(`resourceType must be one of: ${Object.keys(FHIRCAST_RESOURCE_TYPES).join(', ')}`);
```

Comment:
Probably not a `TypeError`

Wondering if we want to be strict here, vs allowing any resource types?  I don't think the spec requires us to be strict, correct?

Reply:
The resource types for the enumerated events are explicit. I think if you want to use other resources you will have to use custom events. I think it's better to enforce them and will be less error prone but I think we can allow extra resources at the end of the context array to allow some flexibility for spec-enumerated events.